ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx

```diff
@@ -38,6 +239,259 @@ export const removeOldDrawing = ({ map, id }: { map: ol.Map; id: string }) => {
+    case 'poly':
+    default:
+      return 'Polygon'
+  }
+}
+
+const getDrawModeFromShape = (shape: Shape): DrawModeType => {
+  switch (shape) {
+    case 'Bounding Box':
+      return 'bbox'
+    case 'Point':
+    case 'Point Radius':
+      return 'circle'
+    case 'Line':
+      return 'line'
+    case 'Polygon':
+    default:
+      return 'poly'
+  }
+}
+
+// see generateAnyGeoFilter in CQLUtils.ts for types
+const getGeoType = (geo: GeometryJSON) => {
+  switch (geo.properties.shape) {
+    case 'Line':
+      return 'LINE'
+    case 'Point':
+      return 'POINT'
+    case 'Point Radius':
+      return 'POINTRADIUS'
+    default:
+      return 'POLYGON'
+  }
+}
+
+const createPolygonModel = (geo: GeometryJSON) => {
+  // Ignore Z coordinate if exists
+  const polygon = (geo.geometry as Polygon).coordinates[0].map((position) =>
+    position.length > 2 ? position.slice(0, 2) : position
+  )
+  return {
+    polygon: Common.wrapMapCoordinatesArray(polygon as any).map((coords) =>
+      coords.map((coord) => Number(coord.toFixed(6)))
+    ),
+    polygonBufferWidth: geo.properties.buffer?.toString() || '0',
+    polygonBufferUnits: geo.properties.bufferUnit,
+  }
+}
+
+const createLineStringModel = (geo: GeometryJSON) => {
+  // Ignore Z coordinate if exists
+  const line = (geo.geometry as LineString).coordinates.map((position) =>
+    position.length > 2 ? position.slice(0, 2) : position
+  )
+  return {
+    line: Common.wrapMapCoordinatesArray(line as any).map((coords) =>
+      coords.map((coord) => Number(coord.toFixed(6)))
+    ),
+    lineWidth: geo.properties.buffer?.toString() || '0',
+    lineUnits: geo.properties.bufferUnit,
+  }
+}
+
+const createPointRadiusModel = (geo: GeometryJSON) => {
+  const wrapped = Common.wrapMapCoordinatesArray([
+    (geo.geometry as Point).coordinates,
+  ] as any)
+  return {
+    lon: Number(wrapped[0][0].toFixed(6)),
+    lat: Number(wrapped[0][1].toFixed(6)),
+    radius: geo.properties.buffer?.toString() || '1',
+    radiusUnits: geo.properties.bufferUnit,
+  }
+}
+
+const createBoundingBoxModel = (geo: GeometryJSON) => {
+  // bbox order: west, south, east, north
+  const wrapped = Common.wrapMapCoordinatesArray([
+    [geo.bbox[0], geo.bbox[1]],
+    [geo.bbox[2], geo.bbox[3]],
+  ])
+  return {
+    west: Number(wrapped[0][0].toFixed(6)),
+    south: Number(wrapped[0][1].toFixed(6)),
+    east: Number(wrapped[1][0].toFixed(6)),
+    north: Number(wrapped[1][1].toFixed(6)),
+  }
+}
+
+const createGeoModel = (geo: GeometryJSON) => {
+  switch (geo.properties.shape) {
+    case 'Polygon':
+      return createPolygonModel(geo)
+    case 'Line':
+      return createLineStringModel(geo)
+    case 'Point':
+    case 'Point Radius':
+      return createPointRadiusModel(geo)
+    case 'Bounding Box':
+      return createBoundingBoxModel(geo)
+    default:
+      return {}
+  }
+}
+
+// TODO do we need to do any polygon 'fixing' here? Or does the validation take care of that?
```

Comment:
Assuming validation covers issues needing fixing

Reply:
Yes, removed this.